docs/language/src/builtins/widgets.md

```diff
@@ -40,6 +40,7 @@ A simple button. Common types of buttons can also be created with [`StandardButt
 -   **`icon`** (_in_ _image_): The image to show in the button. Note that not all styles support drawing icons.
 -   **`pressed`**: (_out_ _bool_): Set to true when the button is pressed.
 -   **`text`** (_in_ _string_): The text written in the button.
+-   **`default`** (_in_ _bool_): Defines the button as default button and display it in a highlighted color.
```

Comment:
This says it displays it in a highlighted color *and* that it defines the button as a "default" button. The documentation here needs to say what that means.

IMO such a default attribute should not only configure the visual appearance but also the semantics: If I choose `default: true;` I want it so that when the user presses "Enter" in a dialog, this button gets activated.

So either the property here configures only the visuals - and then the docs should say just that - or the docs should explain what "default" means and this PR should implement it.

Reply:
Ok, then maybe it would make more sense to rename the property as primary, to define it as visual setting.

Reply:
There is what the Qt docs say: https://doc.qt.io/qt-6/qpushbutton.html#default-prop

Reply:
> There is what the Qt docs say: https://doc.qt.io/qt-6/qpushbutton.html#default-prop

I'm not sure about that behavior. If we have only the `default` property and it`s enable automatically the `default` behavior and you cannot use the button in the highlight/default look and feel without that behavior. Therefore e.g. in my former WPF projects, there was mostly a primary style for buttons and the behavior was defined for the specific context e.g. for dialogs, without to integrate it with the button. @tronical @ogoffart, what do you think. Integrate the default behavior or only integrate a flag to show a button as highlighted in the accent color?

Reply:
FWIW, we could also have a boolean `primary` property now that just covers the visual bit and we implement support for central `Return` handling later. That's kind of what your patch does. What do you think?

I'd be okay with that, as long as the documentation is clear and we track the missing "Return" handling in an issue - so that our conclusion and decision is on record ;)

Reply:
If we implement that feature. Would it make sense to integrate the default/primary property also for TouchArea, then  a developer/ designer can also make use of it for the creation of a custom button in Slint.

Reply:
I'm not sure about `TouchArea` (that seems unrelated to keyboard handling to me), but you're right: We almost have this with the `dialog-button-role`, except that it's only for dialogs and only affects the layout.

Okay, I'm getting convinced that we should go with `primary` as visual-only property for now :)

Reply:
Okay. I have updated the property name to  `primary` and also adjusted the documentation.